src/main/java/it/pagopa/ecommerce/commons/utils/NpgClientUtils.java

```diff
@@ -71,9 +71,12 @@ public enum EndToEndId {
      *         fallback.
      */
     public static String getPaymentEndToEndId(OperationDto operationDto) {
```

Comment:
Looking at the old Kotlin code (https://github.com/pagopa/pagopa-ecommerce-event-dispatcher-service/pull/355/files) and the new null condition, perhaps the whole method could be refactored as follows? it's just an idea :-) 
```
public String getPaymentEndToEndId(OperationDto operationDto) {
    String paymentEndToEndId = null;

    switch (operationDto.getPaymentCircuit()) {
        case NpgClient.PaymentMethod.BANCOMATPAY_SERVICE_NAME:
            paymentEndToEndId = (String) operationDto.getAdditionalData().get(EndToEndId.BANCOMAT_PAY.getValue());
            break;
        case NpgClient.PaymentMethod.MYBANK_SERVICE_NAME:
            paymentEndToEndId = (String) operationDto.getAdditionalData().get(EndToEndId.MYBANK.getValue());
            break;
        default:
            paymentEndToEndId = operationDto.getPaymentEndToEndId();
            break;
    }   
    return paymentEndToEndId;
}

```

Reply:
I would refactoring the whole method using java optionals due the response fields nullability
```suggestion
        public static String getPaymentEndToEndId(OperationDto operationDto) {
        String paymentEndToEndId = Optional.ofNullable(operationDto).map(OperationDto::getPaymentEndToEndId).orElse(null);
        return Optional
                .ofNullable(operationDto)
                .filter(op -> op.getPaymentCircuit() != null)
                .flatMap(op ->
                        Optional.ofNullable(op.getAdditionalData())
                )
                .map(additionalData ->
                        switch (NpgClient.PaymentMethod.fromServiceName(operationDto.getPaymentCircuit())) {
                            case BANCOMATPAY -> additionalData.get(EndToEndId.BANCOMAT_PAY.getValue()).toString();
                            case MYBANK -> additionalData.get(EndToEndId.MYBANK.getValue()).toString();
                            case default -> paymentEndToEndId;
                        }
                )
                .orElse(paymentEndToEndId);
    }
```

this could be an alternative implementation that could be a bitter easier to read and that use tu functional approach used in other classes (N.B. this approach require to add a getter method on the EndToEndId enumeration to retrieve the associated value)

Reply:
Thanks. Done here: 6f427cc7